lib/hocon/impl/config_reference.rb

```diff
@@ -72,4 +73,56 @@ def resolve_substitutions(context, source)
     end
 
   end
+
+  def self.not_resolved
```

Comment:
Should this be static? It looks like it's not in the Java library, and it uses the object on which it is being called.